### PR TITLE
Add preview links for profile backgrounds in the points shop

### LINKED ISSUE #1063: [FEATURE REQUEST] Preview background for item shop
### Current Behavior (if any)
Item shop is lacking the preview background feature

### Desired Behavior
Add a button to preview background for (animated) background available in the steam item shop

### Motivation / Use Case for Changing the Behavior
More convenient and consistent with market feature.

### PR DESCRIPTION
**Pull request by candela97**
Closes #1063.

### CODE REVIEW

File: src/js/Content/Features/Store/PointsShop/FBackgroundPreviewLink.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    new MutationObserver((mutations, observer) => {
+                        videoEl = mutations[0].addedNodes[0].querySelector("[class^='redeempointsmodal_BackgroundPreviewContainer'] video");
+                        observer.disconnect();
+                        resolve();
+                    }).observe(addedNodes[0], {"childList": true, "subtree": true});
+                });
+
+                if (!videoEl) { continue; }
+
+                const bgLink = videoEl.querySelector("source").src.match(/images\/items\/(\d+)\/([a-z0-9.]+)/i);
```
Comment (candela97) [inline src/js/Content/Features/Store/PointsShop/FBackgroundPreviewLink.js:28]: To keep it simple, only use the first source (`webm` format, should be well supported)

Comment (candela97) [inline src/js/Content/Features/Store/PointsShop/FBackgroundPreviewLink.js:28]: To keep it simple, only use the first source (usually `webm` format, should be well supported)

File: src/js/Content/Features/Store/PointsShop/FBackgroundPreviewLink.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+                let videoEl;
+
+                await new Promise(resolve => {
+                    new MutationObserver((mutations, observer) => {
+                        videoEl = mutations[0].addedNodes[0].querySelector("[class^='redeempointsmodal_BackgroundPreviewContainer'] video");
+                        observer.disconnect();
+                        resolve();
+                    }).observe(addedNodes[0], {"childList": true, "subtree": true});
+                });
```
Comment (candela97) [inline src/js/Content/Features/Store/PointsShop/FBackgroundPreviewLink.js:24]: Is it possible the modal content is loaded before the observer is attached? This worked while testing so I assumed no...

**Comment by MxtOUT** [comment]
I only see a static background when previewing

**Comment by candela97** [comment]
Did you disable autoplaying videos? In that case you'll have to manually play the video. Not sure what to do about that, since if autoplay is disabled, animated backgrounds are just not gonna autoplay, preview or not.

**Comment by MxtOUT** [comment]
I don't know what caused that but I still had autoplay disabled and this time it worked.
But I experienced in the past that this whole autoplay thing doesn't work reliably, so it's fine

**Comment by candela97** [comment]
Something to fix: the feature doesn't work when viewing the item directly, e.g. https://store.steampowered.com/points/shop/c/backgrounds/reward/150370/, probably due to our script running too late. You can get these links by clicking on the icon next to the item name in the popup dialog.

**Comment by candela97** [comment]
Something to fix: the feature doesn't work when viewing the item directly, e.g. https://store.steampowered.com/points/shop/c/backgrounds/reward/150370/, probably due to our script running too late. You can get these links by clicking on the icon next to the item name in the popup dialog.

**Pull request merged**